src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -314,7 +316,7 @@ def adj_op_direct(self, coeffs, image=None, grid_data=False):
                 order="F",
             )
         self.operator.adj_op_direct(coeffs.data.ptr, image.data.ptr, grid_data)
-        image = image.reshape(C, *self.shape[::-1])
+        image = image.T.reshape(C, *self.shape[::-1], order="C")
```

Comment:
This seems to have been the issue.
`<>.T.reshape(..., order="C")` is not the same as `<>.reshape(..., order="F")`